app/modules/asset.rb

```diff
@@ -0,0 +1,148 @@
+class Asset
+
+  DEFAULT_START_DATE = 8.days.ago.strftime('%F')
+  DEFAULT_END_DATE = 1.day.ago.strftime('%F')
+
+
+  attr_reader :basic_stats,
+              :sources,
+              :os_sources,
+              :traffic_sources,
+              :country_sources,
+              :service,
+              :params,
+              :property,
+              :current_start_date,
+              :current_end_date,
+              :previous_period,
+              :current_period
+
+
+  def initialize(service, params)
+    @service = service
+    @params = params
+    @property = get_property
+    @basic_stats = get_basic_stats
+    @sources = get_sources
+    @os_sources = get_os_sources
+    @traffic_sources = get_traffic_sources
+    @country_sources = get_country_sources
+    @previous_pariod, @current_period = basic_stats.rows.each_slice(basic_stats.rows.size/2).to_a
+    @current_start_date = set_date(current_period[0][0])
+    @current_end_date = set_date(current_period[-1][0])
+  end
+
+
+  def change_in_percent
+    "#{change}%"
+  end
+
+  def change_status
+    change.positive? ? 'increase' : 'decrease'
+  end
+
+  def current_visits
+    summarize_visits(@current_period)
+  end
+
+  def previous_visits
+    summarize_visits(@previous_pariod)
+  end
+
+  def current_page_views
+    summarize_page_views(@current_period)
+  end
+
+  def previous_page_views
+    summarize_page_views(@previous_pariod)
+  end
+
+  def image
+    ReportGenerator.generate(self)
+  end
+
+  private
+
+  def summarize_visits(period)
+    period.inject(0) { |sum, e| sum + e[1].to_i }
+  end
+
+  def summarize_page_views(period)
+    period.inject(0) { |sum, e| sum + e[2].to_i }
+  end
+
+  def change
+    (((current_visits.to_f / previous_visits) - 1)*100).round(1)
+  end
+
+  protected
+
+  def set_date(index)
+    Date.parse(index)
+  end
+
+  def get_property
+    @service.get_web_property(@params[:account_id],
+                              @params[:web_property_id])
+  end
+
+  def get_basic_stats
+    profile_id = "ga:#{@params[:profile_id]}"
+    start_date = (Date.parse(DEFAULT_START_DATE) - 6.days).strftime('%F')
+    end_date = DEFAULT_END_DATE
+    metrics = 'ga:sessions, ga:uniquePageviews'
```

Comment:
a lot of hardcoded `ga: ` strings around, that makes us pretty coupled to Google Analytics. Now that is probably fine, but I just want to check if we are content with that coupling.